torch_geometric/datasets/ba_shapes.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional, Callable
+
+import torch
+import numpy as np
+import networkx as nx
+from torch_geometric.data import InMemoryDataset, Data
+
+
+def house():
+    G = nx.house_graph()
+    labels = [1, 1, 2, 2, 3]
+    attr = {i: labels[i] for i in range(5)}
+    nx.set_node_attributes(G, attr, 'label')
+    nx.set_edge_attributes(G, 1, 'label')
+    return G
+
+
+def ba(num_nodes=300, m=5, seed=None):
+    G = nx.barabasi_albert_graph(num_nodes, m, seed=seed)
+    nx.set_node_attributes(G, 0, 'label')
+    nx.set_edge_attributes(G, 0, 'label')
+    return G
+
+
+class BAShapes(InMemoryDataset):
+    r"""The BA-Shapes dataset from the `"GNNExplainer: Generating Explanations
+    for Graph Neural Networks" <https://arxiv.org/pdf/1903.03894.pdf>`_ paper,
+    containing a Barabasi-Albert (BA) graph with 300 nodes and a set of 80
+    "house"-structured graphs connected to it.
+
+    Args:
+        seed (int, optional): Random seed used for :obj:`networkx` Barabási
```

Comment:
Can we also add the option to load the `BAShapes` data pre-defined from `PGExplainer`?